src/controllers/AuthenticationController.ts

```diff
@@ -0,0 +1,345 @@
+      });
+      const payload: Payload = {
+        id: createdUser.id,
+        role: createdUser.role,
+        schoolId: createdUser.schoolId,
+        isConfirmed: createdUser.isConfirmed,
+        isTemporary: createdUser.isTemporary,
+      };
+      const accessToken = JWTUtils.generateAccessToken(payload, undefined, {});
+      // TODO: insert frontend url for confirm email.
+      const url = `test.com?token=${accessToken}`;
+      await this.emailService.sendConfirmEmail(email, url);
+      res.json({
+        message: userFriendlyMessage.success.createUser,
+      });
+    } catch (e) {
+      res.status(400);
+      res.json({message: userFriendlyMessage.failure.createUser});
+      next(e);
+    }
+  }
+
+  async confirmEmail(req: Request, res: Response, next: NextFunction) {
+    try {
+      const token = req.query.token as string;
+      let decoded;
+      try {
+        decoded = JWTUtils.verifyAccessToken(token) as Payload;
+      } catch (e) {
+        res.status(401);
+        res.json({message: userFriendlyMessage.failure.invalidToken});
+        return;
+      }
+      const {id} = decoded;
+      const user = await this.userService.getOneUserById(id);
+
+      if (!user) {
+        res.status(400);
+        res.json({message: userFriendlyMessage.failure.userNotExist});
+      }
+
+      const updatedAttributes: UserAttributes = {
+        ...user,
+        isConfirmed: true,
+      };
+      await this.userService.updateOneUserById(id, updatedAttributes);
+      res.json({message: userFriendlyMessage.success.confirmEmail});
+      // Redirect user to login page
+    } catch (e) {
+      res.status(400);
+      res.json({message: userFriendlyMessage.failure.confirmEmail});
+      next(e);
+    }
+  }
+
+  // Format for CSV file: [email], [role]
+  // TODO: Output CSV containing [email], [role], [url]
+  async bulkSignUp(req: Request, res: Response, next: NextFunction) {
+    try {
+      const userAttributes = req.parsedUserAttributes;
+
+      if (!userAttributes) {
+        res.status(400);
+        res.json({message: userFriendlyMessage.failure.userAttributes});
+        return;
+      }
+
+      //TODO: Auth Middleware will add the current user in req.user
+      const {user} = req;
+      const userCreationAttributes: UserCreationAttributes[] = [];
+
+      for (const attribute of userAttributes) {
+        const email = attribute[0];
+        const user = await this.userService.getOneUserByEmail(email);
+
+        if (user) {
+          res.status(400);
+          res.json({
+            email: email,
+            message: userFriendlyMessage.failure.emailExists,
+          });
+          return;
+        }
+        userCreationAttributes.push({
+          email: attribute[0],
+          password: PasswordUtils.generateRandomPassword(),
+          role: attribute[1] as Role,
+          schoolId: 1,
+          // TODO: Uncomment below after implementation of auth middleware
+          // schoolId: user.schoolId,
```

Comment:
user here will refer to user in line 119, when you intended for it for it to refer to the user in line 114. Scope error. You will need to change one of your variable names.